### DISCUSSION TITLE
Add `value` property to be passed to fields in `APIPlayground`

### DISCUSSION THREAD

**Discussion by olabrahmi** [fuma-nama/fumadocs#643] [Ideas]
is there any possibility to allow for fully controlling the value of inputs such as authorization in `APIPlayground` component? A use case for this is to be able to populate an access token, email, name, or other info and keep it in sync with state/local storage.

current `PrimitiveRequestField` field only contains `defaultValue` currently, a `value` prop would also be helpful in such cases.

```ts
interface PrimitiveSchema extends BaseSchema {
    type: 'boolean' | 'string' | 'number';
    defaultValue?: string;
    value?: string;
}
```

**Comment by fuma-nama** [comment]
To make it controlled, you will need onChange and other props while it's using react-hook-form. I think it would be more appropriate to support passing a custom component for authentication fields(?) so you can add custom buttons (e.g. login) and use the user token directly for API playground

**Reply by olabrahmi** [reply]
That would work great too. Wouldn't it make more sense to allow that for all fields too, including body? There are cases where one would like to populate those with a user's info (email, for instance) as I pointed out before.

And one more thing, if something like this is to be added, could there be a way to tweak it in the `generateDocs` OpenAPI generator? Correct me if I'm wrong, but currently it doesn't seem to support any (advanced?) customizations such as passing custom props, change order of components, tweak output MDX content...

**Reply by fuma-nama** [reply]
You have to do it by replacing imports using [Imports](https://fumadocs.vercel.app/docs/ui/openapi#imports), and import your own component to pass custom props.
